cmd/alertmanager/main.go

```diff
@@ -481,16 +482,25 @@ func clusterWait(p *cluster.Peer, timeout time.Duration) func() time.Duration {
 	}
 }
 
-func extURL(listen, external string) (*url.URL, error) {
+type hostnameFunc func() (string, error)
```

Comment:
What is this indirection through `extURL -> hostnameFunc -> resolveExtURL` for? I don't understand why this is not just a single function.

Reply:
To resolve the external URL we need to get the hostname but I didn't want to call `os.Hostname` in the tests so I went for this convoluted way. I'll add a comment unless you can think of something clearer.

Reply:
How about passing `os.Hostname` as an argument to `resolveExtURL`? Would remove the complexity of instantiating a new type.